src/libraries/System.Reflection/tests/ILLink.Descriptors.xml

```diff
@@ -0,0 +1,6 @@
+<linker>
+  <assembly fullname="System.Private.CoreLib">
+    <type fullname="System.Collections.ArrayList" />
```

Comment:
Why is this necessary?

Reply:
[`e09cad4` (#51697)](https://github.com/dotnet/runtime/pull/51697/commits/e09cad4d691dcb66c837739fc1ca59d84e4226c0)

```
    `IList` implementation on `ArrayList` was getting trimmed

   
      fail: [FAIL] System.Reflection.Tests.TypeInfoTests.IsAssignable(type: typeof(System.Collections.IList), c: typeof(System.Collections.ArrayList), expected: True)
      info: Assert.Equal() Failure
      info: Expected: True
      info: Actual:   False
      info:    at System.Reflection.Tests.TypeInfoTests.IsAssignable(Type type, Type c, Boolean expected) in /Users/radical/dev/r3/src/libraries/System.Reflection/tests/TypeInfoTests.cs:line 580
      info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/radical/dev/r3/src/mono/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:line 378
    ```

Reply:
If it is just that one test, maybe adding a `DynamicDependency` on the test would be a better approach. Then it localizes it, and you can tell why it is needed. Having it rooted here makes it hard to reason about.